dist/tools/pic32prog/doc.md

```diff
@@ -0,0 +1,158 @@
+Flashing on Linux using pickit-3
+================================
+
+This document describes how to enable flashing a pic32-wifire from Linux with a
+PICkit3 using [pic32prog](https://github.com/sergev/pic32prog).
+
+It will require flashing a specific firmware on the PICkit3.
+As this can only be done from a Windows computer, that not many Linux users
+have, the following steps explain how to setup a Windows VirtualBox virtual
+machine and flash the PICkit3 from it.
+
+Informations come from this comment
+
+https://github.com/RIOT-OS/RIOT/pull/6092#issuecomment-261987955
+
+
+Steps
+-----
+
+### Setup your computer
+
+I ran this on Ubuntu 16.04
+
+* Install Virtualbox
+* Add yourself to the 'vboxusers' group
+  `sudo usermod -a -G vboxusers your_username`
+* Reboot
+
+### Setup Windows VM
+
+You can download a free 90 days valid windows virtualbox image from:
+
+    https://developer.microsoft.com/en-us/microsoft-edge/tools/vms/
+
+I downloaded a Windows 10 virtualbox image
+
+* Create a machine in virtualbox in File-Import Appliance.
+* Start the virtual machine and wait until you are on the windows desktop
+* Install Virtualbox Guest-additions for USB support
+  * Doc: https://www.virtualbox.org/manual/ch04.html#additions-windows
+* Shutdown the virtual machine
+* In the virtual machine settings, go to USB:
+  * Select "Enable USB Controller" with "USB 1.1"
+  * Click the small '+' sign on the right and add the Microchip devices.
+  * Doc: https://www.virtualbox.org/manual/ch03.html#idm1640
+* Start the virtual machine
+
+### Downloads
+
+Download and extract:
+
+* PICkit3-Programmer http://ww1.microchip.com/downloads/en/DeviceDoc/PICkit3%20Programmer%20Application%20v3.10.zip
+
+Adapted from the README (on Windows 10 machine .NET 4.00 is already installed)
+
+* Extract `PICkit3 Programmer Application Setup v3.10.zip`
+* Run `setup.exe`
+* Run PICkit3
+* Tools/Download PICkit Operating System
+  * Select `c://Program Files/Microchip/PICkit3/PK3OSV020005.hex`
+* Device should be flashed there will be a message saying the board is found
+* Close the program (it crashes when closing it but it is ok)
+* Turn off your VM
+
+
+pic32prog
+---------
+
+Download and compile `pic32prog` flasher
+
+https://github.com/sergev/pic32prog
+
+Run it without the board connected
+
+```
+$ pic32prog
+Programmer for Microchip PIC32 microcontrollers, Version 2.0.221
+    Copyright: (C) 2011-2015 Serge Vakulenko
+      Adapter: PICkit3 Version 2.0.5
+No device attached.
+
+No target found.
+```
+You should see the `Adapter: PICkit3 Version 2.0.5` line.
+
+### Troubleshooting
+
+If you do not get the `Adapter: PICkit3 Version 2.0.5` line,
+try running it with `sudo pic32prog`. If it works with `sudo` it is a
+permission issue.
+
+Add yourself to the `plugdev` group, add the following `udev` rule and reboot.
+(Find a tutorial if you do not know how to do this).
```

Comment:
RTFM is not very friendly. Especially since explaining that you need to add the rule to a file in `/etc/udev/rules.d` is just as short ;-)

Reply:
Yeaaaaahhh you are right.

Reply:
I checked my files and found that I had mplab udev rules, that is why it was detected on my computer.

They had some wildcard values for `idProduct` and found this other website talking about rules for these devices: http://en.microstickplus.com/mplabx-on-linux
So I put the link and only use `idVendor` as selection.